README.md

```diff
@@ -67,14 +69,14 @@ SSB-JS organization require discussion and consent.
 - 'Maintainers' are contributors trusted to steward a project and vote.
 - Maintainers MUST have the 'owner' role on GitHub and NPM.
 - Maintainers MUST have 2FA enabled on GitHub and NPM.
+- Maintainers MUST communicate in the project's CONTRIBUTING.md what types of changes in a project pull request are blocked from merging.
```

Comment:
Clarification: Could you add some details on what this might look like? I'm having a hard time imagining writing down all the things that might block a PR, and I'm wondering whether this might already be covered in the next point: "communicate [...] what is expected in a pull request".

Reply:
That's interesting! If you feel like many things could block a PR, that's a good sign we should write them down explicitly as opposed to keeping it implicit.

The "things that are expected in a PR" may not suffice to define blocking concerns. E.g. what if the PR contains 400 commits, only 1 of those commits fixes the issue at hand. At the same time, code style is changed everywhere, node_modules is committed into git, and the LICENSE is deleted. The thing we expected, the fix, **is** there, but so are a bunch of unexpected things.

In practice, contributing.md could list things like:

"We won't accept PRs that do too many changes at once, or do side effects unrelated to the issue at hand. We won't accept PRs that create new security vulnerabilities." Etc

Reply:
> The thing we expected, the fix, is there, but so are a bunch of unexpected things.

C4 solves this problem by identifying a patch as "a minimal and accurate answer to exactly one identified and agreed problem". I think we should write down the rules we frequently enforce, especially if we can write them as tests, but we should also give maintainers the ability to say "we haven't written rules about this yet, but I'm requesting changes".

The number of potential problems is infinite, and I think we'd be better off with:

- A general expectation of what patches should look like.
- Tests that a user can run locally to ensure that their patch conforms.
- The ability for maintainers to say "please change X" without requiring that it's already written in the contributor guide.

In other words: I'm worried that this rule forces maintainers to merge anything that doesn't break an already-written-down rule.

```suggestion
- Maintainers SHOULD communicate in the project's CONTRIBUTING.md what types of changes in a project pull request are frequently blocked from merging.
```